src/main/java/uk/gov/digital/ho/hocs/casework/migration/api/MigrationStageService.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.digital.ho.hocs.casework.migration.api;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.digital.ho.hocs.casework.api.CaseworkConstants;
+import uk.gov.digital.ho.hocs.casework.api.dto.CreateStageRequest;
+import uk.gov.digital.ho.hocs.casework.client.auditclient.AuditClient;
+import uk.gov.digital.ho.hocs.casework.domain.model.BaseStage;
+import uk.gov.digital.ho.hocs.casework.domain.model.CaseData;
+import uk.gov.digital.ho.hocs.casework.domain.model.Stage;
+import uk.gov.digital.ho.hocs.casework.domain.repository.StageRepository;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import static net.logstash.logback.argument.StructuredArguments.value;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.EVENT;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.STAGE_CREATED;
+
+@Slf4j
+@Service
+public class MigrationStageService {
+
+    private final StageRepository stageRepository;
+    private final AuditClient auditClient;
+    private final MigrationCaseDataService migrationCaseDataService;
+
+    public MigrationStageService(StageRepository stageRepository,
+                                 AuditClient auditClient,
+                                 MigrationCaseDataService migrationCaseDataService) {
+        this.stageRepository = stageRepository;
+        this.auditClient = auditClient;
+        this.migrationCaseDataService = migrationCaseDataService;
+    }
+
+    @Transactional
+    public Stage createStageForClosedCase(UUID caseUUID, String stageType) {
+
+        log.debug("Creating Stage: {} for case {}", stageType, caseUUID);
+        CreateStageRequest createStageRequest = new CreateStageRequest(stageType, null, null, "", null, null);
+
+        Set<Stage> allStages = stageRepository.findAllByCaseUUIDAsStage(caseUUID);
```

Comment:
This won't be needed for a closed case as a stage won't exist at this point.

Reply:
Removed